test/test-with-server.js

```diff
@@ -235,6 +236,21 @@ describe("follow-redirects", function () {
       });
   });
 
+  it("should callback setTimeout on the returned stream", function () {
```

Comment:
we should also test the case where the timeout limit is not reached

Reply:
will do